user_home/views.py

```diff
@@ -0,0 +1,105 @@
+from django.contrib import messages
+from django.contrib.auth import logout
+from django.core.exceptions import ObjectDoesNotExist
+from django.forms.models import model_to_dict
+from django.http import HttpRequest, JsonResponse
+from django.shortcuts import redirect, render
+
+from .forms import NewTaskForm
+from .models import Task
+
+
+def main(request: HttpRequest):
+    if not request.user.is_authenticated:
+        return redirect("/")
+    user = request.user
+    tasks = user.task_set.all()
+    return render(request, "main.html", {"title": "Main", "user_tasks": tasks})
+
+
+def signout(request: HttpRequest):
+    if not request.user.is_authenticated:
+        return redirect("/")
+    logout(request)
+    return redirect("/")
+
+
+def new_task(request: HttpRequest):
```

Comment:
it's also better to separate functions, get form, and post form.

1 function shouldn't handle both cases.